M2/Macaulay2/d/evaluate.d

```diff
@@ -1670,6 +1670,18 @@ testfun(c:Code):Expr := (
     when r is Error do r else nullE);
 setupop(TestS, testfun);
 
+export toExpr(err:Error):Expr := Expr(SpecialExpr(Class(err), Expr(err)));
+
+attemptfun(c:Code):Expr := (
+    r := tryEval(c);
+    if tryEvalSuccess then seq(r, nullE)
+    else (
+	when r
+	is err:Error do seq(nullE, toExpr(err))
+	-- shouldn't happen since tryEvalSuccess only false when r is an Error
+	else buildErrorPacket("unable to catch error")));
+setupop(attemptS, attemptfun);
```

Comment:
Now that I look at how simple this is, could you remind me what's the issue or difficulty with this syntax?
```m2
try (...) catch err do (...) -- assigns the error object to `err`.
```
This might satisfy all parties.

(If this doesn't work, I think renaming to `trap` will be good, so the error can say `unable to trap the error`. Also, the indentation looks strange. You could do `else when r is ...`)

Reply:
That's probably doable.  We'd probably need to introduce a new `TryCatchDo` member of the `ParseTree` union, but that shouldn't be too bad.  Would that be the only combination, or are you envisioning `else` and/or `then` showing up with `catch` and `do` as well? 

I am a little worried that `catch` may have too high of a precedence -- we'd probably want whatever keyword went here to have a low one matching `do`, `else`, and `then`.  Maybe `try ... trap ... do`?

Reply:
I've switched it to `try ... except err do ...`